### PR TITLE
Fix monitor state_averager in examples/albert (preliminary)

### PR DESCRIPTION
**Pull request by justheuristic**
This PR fixes several minor issues found in #446 : 

- fix `prefix=...` in training monitor
- create scheduler in training monitor
- rename experiment_prefix -> run_id
- enable checkpoints on aux peer by default
- decouple total steps from scheduler max steps

Tests:
- [x] train for a few hundred steps
- [x] download state from aux peer
image

In a subsequent PR:
- __create optimizer normally in training monitor (instead of TrainingStateAverager)__
- stream data from wikitext instead of preprocessing it
- extract common code from trainer and monitor to utils
- allow training monitor to assist in averaging
- refer to advanced example

### CODE REVIEW

File: examples/albert/README.md [hunk 1]
```diff
@@ -55,9 +55,7 @@ To join the collaboration with a GPU trainer,
   (see [default paths](./arguments.py#L117-L134) for reference)
 - Run:
   ```bash
-  ./run_trainer.py \
-      --initial_peers ONE_OR_MORE_PEERS \
-      --logging_first_step --output_dir ./outputs --overwrite_output_dir --logging_dir ./logs
+  ./run_trainer.py  --initial_peers ONE_OR_MORE_PEERS --per_device_train_batch_size BATCH_SIZE_FOR_YOUR_GPU_DEFAULT_4
```
Comment (borzunov) [inline examples/albert/README.md:58]: nit:

```suggestion
  ./run_trainer.py  --initial_peers ONE_OR_MORE_PEERS --per_device_train_batch_size BATCH_SIZE_FOR_YOUR_GPU
```

Because (1) `BATCH_SIZE_FOR_YOUR_GPU_DEFAULT_4` is hard to parse and (2) we're talking about the default value below.

File: examples/albert/arguments.py [hunk 2]
```diff
@@ -138,9 +138,14 @@ class AlbertTrainingArguments(TrainingArguments):
...
+    output_dir: str = "outputs"
     logging_steps: int = 100
+    logging_first_step: bool = True
+    overwrite_output_dir: bool = True
+
     save_total_limit: int = 2
     save_steps: int = 500
-
-    output_dir: str = "outputs"
+    max_steps: int = 10 ** 30
```
Comment (borzunov) [inline examples/albert/arguments.py:151]:
```suggestion
```

Let's do of the options below:

(1) remove it (it's inf by default AFAIU)
(2) set it to a reasonable value
(3) add a comment that you mean `inf` by writing `10 ** 30` (now, it's not 100% clear what this number means, someone may think that you really need that much microsteps)

**Review by borzunov — APPROVED** [review]
Awesome PR!

File: examples/albert/run_training_monitor.py [hunk 3]
```diff
@@ -40,6 +40,7 @@ class TrainingMonitorArguments(BaseTrainingArguments):
     wandb_project: Optional[str] = field(
         default=None, metadata={"help": "Name of Weights & Biases project to report the training progress to"}
     )
+    store_checkpoints: bool = field(default=True, metadata={"help": "If False, disables CheckpointHandler altogether"})
```
Comment (mryab) [inline examples/albert/run_training_monitor.py:43]:
```suggestion
    store_checkpoints: bool = field(default=True, metadata={"help": "If False, disables periodic checkpoint saving"})
```

CheckpointHandler is an internal detail of this script which might not be known to the user

File: examples/albert/arguments.py (hunk 2 above)
Comment (mryab) [inline examples/albert/arguments.py]: Is there a point in continuing training after `total_steps` have passed? It seems that with the current schedule, the learning rate for all subsequent updates will be 0

File: examples/albert/run_training_monitor.py [hunk 4]
```diff
@@ -99,7 +99,8 @@ def __init__(
         self.state_averager = TrainingStateAverager(
             dht=dht,
             optimizer=opt,
-            prefix=experiment_prefix,
+            scheduler=get_linear_schedule_with_warmup(opt, num_warmup_steps=5000, num_training_steps=125_000),
```
Comment (mryab) [inline examples/albert/run_training_monitor.py:102]: It's better to replace these values with the defaults from the config to avoid possible divergence in the future and general "magic constant" readability issues. The best way might be to move the number of training and warmup steps to a common config used both in the monitor and the trainer code

File: examples/albert/arguments.py (hunk 2 above)
Comment (justheuristic) [reply examples/albert/arguments.py]: went with a 3

File: examples/albert/run_training_monitor.py (hunk 4 above)
Comment (borzunov) [reply examples/albert/run_training_monitor.py:102]: I think we shouldn't just move args to a common config, but move the whole code creating `hivemind.Optimizer` to a common file. Otherwise, it's likely that any model/optimizer/scheduler changes made by a user will lead to a divergence between monitors and trainers

File: examples/albert/arguments.py (hunk 2 above)
Comment (justheuristic) [reply examples/albert/arguments.py]: it is, but I don't get what do you mean by this

File: examples/albert/run_training_monitor.py (hunk 4 above)
Comment (justheuristic) [reply examples/albert/run_training_monitor.py:102]: This dataclass is currently not available to aux peer. For long-term optimal composition, please see CALM and/or petabert examples

File: examples/albert/arguments.py (hunk 2 above)
Comment (borzunov) [reply examples/albert/arguments.py]: What's the point of decoupling these params?

Comment (justheuristic) [reply examples/albert/arguments.py]: Ah, I see the problem.

max_steps is the number of local steps, not global optimizer updates. The reason why previous version was incorrect is because peers would run for 125k local forward-backward cycles which would be much less than 125k scheduler epochs

Comment (borzunov) [reply examples/albert/arguments.py]: I think it's still unclear that (1) total_steps and max_steps are measured in different units and (2) the difference between them is still elusive for a user.

So I'm for removing max_steps completely (maybe in a future PR since this may require you to test the scripts again)

Comment (justheuristic) [reply examples/albert/arguments.py]: I've added explanations to each parameter to be safe
